tests/integration/test_command.py

```diff
@@ -50,6 +53,62 @@ def run(cmd: list[str], env: dict[str, str] | None = None) -> subprocess.Complet
     return subprocess.run(cmd, capture_output=True, text=True, check=False, env=env)
 
 
+@cache
+def has_pure_docker() -> bool:
+    """Check if pure working Docker is available.
+
+    Returns:
+        True if Docker is installed, working and not using podman backend.
+    """
+    # https://github.com/podman-desktop/podman-desktop/issues/10745
+    failure = ""
+    default_docker_socket = Path("/var/run/docker.sock")
+    if not shutil.which("docker"):
+        failure = "Docker is not installed."
+    elif default_docker_socket.exists():
+        # detect if docker is NOT using podman's backend (libpod)
+        result = run(
+            ["curl", "--silent", "--unix-socket", "/var/run/docker.sock", "http:/v1/_ping"],
+        )
+        if result.returncode != 0:
+            failure = "Docker does not seem to respond to default socket."
+        else:
+            result = run(
+                [
+                    "curl",
```

Comment:
only because I was curious, this is neat: https://www.hackitu.de/urllib_uds/

Reply:
In fact I wanted to do something like this but decided to postpone. I bookmarked that code and will use it. Likely I will implement this as a fixture inside pytest-plus, so we can reuse it in multiple places.